content/en/docs/reference/networking/virtual-ips.md

```diff
@@ -237,8 +237,13 @@ uses packet processing logic (such as Linux iptables) to define _virtual_ IP
 addresses which are transparently redirected as needed.
 
 When clients connect to the VIP, their traffic is automatically transported to an
-appropriate endpoint. The environment variables and DNS for Services are actually
-populated in terms of the Service's virtual IP address (and port).
+appropriate endpoint. Actually, the kubelet on each node populates environment variables
+for containers based on the port number for the Service and on the virtual IP address
+(_cluster IP_), if any, that has been assigned.
+The cluster DNS component also looks up the port and the virtual IP address for each
```

Comment:
DNS does not care about the ports, unless you are using SRV records, but that is not common.

You may want to reference https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#services ?